zappa/utilities.py

```diff
@@ -360,8 +360,12 @@ def status(self, function):
                     response = self._lambda.call("get_event_source_mapping", UUID=self._get_uuid(function))
                     LOG.debug(response)
                 except botocore.exceptions.ClientError:
-                    LOG.debug("event source %s does not exist", self.arn)
-                    response = None
+                    # https://github.com/zappa/Zappa/issues/1317
+                    if e.response['Error']['Code'] == 'ResourceNotFoundException':
+                        LOG.debug("event source %s does not exist", self.arn)
+                        response = None
+                    else:
+                        raise e
```

Comment:
This `raise` changes the current operation of the `status()` function. I'm not sure if it's desired to raise an exception here.